controller/rest/remote_export.go

```diff
@@ -0,0 +1,249 @@
+	"github.com/neuvector/neuvector/share"
+	log "github.com/sirupsen/logrus"
+)
+
+///////////////////////////////////////////////////////////
+// Remote Export Repository REST Handlers /////////////////
+///////////////////////////////////////////////////////////
+
+func handlerRemoteExportRepositoryPost(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.WithFields(log.Fields{"URL": r.URL.String()}).Debug()
+	defer r.Body.Close()
+
+	acc, login := getAccessControl(w, r, "")
+	if acc == nil {
+		return
+	} else if !acc.Authorize(&share.CLUSRemoteExportRepository{}, nil) {
+		restRespAccessDenied(w, login)
+		return
+	}
+
+	body, _ := ioutil.ReadAll(r.Body)
+	var remoteExportRepository api.REST_RemoteExportRepository_POST
+	err := json.Unmarshal(body, &remoteExportRepository)
+	if err != nil {
+		msg := fmt.Sprintf("Could not unmarshal request body: %s", err.Error())
+		restRespErrorMessage(w, http.StatusBadRequest, api.RESTErrInvalidRequest, msg)
+		return
+	} else if !isObjectNameValid(remoteExportRepository.Alias) {
+		e := "Invalid characters in alias"
+		log.WithFields(log.Fields{"alias": remoteExportRepository.Alias}).Error(e)
+		restRespErrorMessage(w, http.StatusBadRequest, api.RESTErrInvalidRequest, e)
+		return
+	}
+
+	clusRemoteExportRepository := share.CLUSRemoteExportRepository{
+		RepositoryOwnerUsername:          remoteExportRepository.RepositoryOwnerUsername,
+		RepositoryName:                   remoteExportRepository.RepositoryName,
+		RepositoryBranchName:             remoteExportRepository.RepositoryBranchName,
+		PersonalAccessToken:              remoteExportRepository.PersonalAccessToken,
+		PersonalAccessTokenCommitterName: remoteExportRepository.PersonalAccessTokenCommitterName,
+		PersonalAccessTokenEmail:         remoteExportRepository.PersonalAccessTokenEmail,
+	}
+
+	err = clusHelper.CreateRemoteExportRepository(remoteExportRepository.Alias, &clusRemoteExportRepository, nil)
+	if err != nil {
+		msg := fmt.Sprintf("Could not save remote export repository to kv store: %s", err.Error())
+		restRespErrorMessage(w, http.StatusInternalServerError, api.RESTErrFailWriteCluster, msg)
+		return
+	}
+
+	restRespSuccess(w, r, nil, acc, login, &remoteExportRepository, "Create Remote Export Repository")
+}
+
+func handlerRemoteExportRepositoryGet(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.WithFields(log.Fields{"URL": r.URL.String()}).Debug()
+	defer r.Body.Close()
+
+	acc, login := getAccessControl(w, r, "")
+	if acc == nil {
+		return
+	} else if !acc.Authorize(&share.CLUSRemoteExportRepository{}, nil) {
+		restRespAccessDenied(w, login)
+		return
+	}
+
+	alias := ps.ByName("alias")
+	clusRemoteExportRepository, _, err := clusHelper.GetRemoteExportRepository(alias)
+	if err != nil {
+		msg := fmt.Sprintf("Could not retrieve remote export repository from kv store: %s", err.Error())
+		restRespErrorMessage(w, http.StatusInternalServerError, api.RESTErrFailWriteCluster, msg)
```

Comment:
In other API handlers, they simply call
restRespNotFoundLogAccessDenied(w, login, err)
when the method fails to get the object based on the name/alias.
I wonder why it uses http.StatusInternalServerError/api.RESTErrFailWriteCluster here?

Reply:
The `http.StatusInternalServerError` & `api.RESTErrFailWriteCluster` is a mistake here, changed in a recent commit.